### PR TITLE
Feature mes 2032 provide dangerous commentary

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description and relevant Jira numbers

- dangerous comment textarea added to office view
- reducer and action added for the office component

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

Screenshot 2019-04-15 at 13 49 05

### CODE REVIEW

File: src/pages/office/office.actions.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import { Action } from '@ngrx/store';
 
 export const OFFICE_VIEW_DID_ENTER = '[OfficePage] Office view did enter';
+export const OFFICE_VIEW_DID_ADD_DANGEROUS_FAULT_COMMENT = '[OfficePage] Office view did add dangerous fault comment';
```
Comment (johnmorleybjss) [inline src/pages/office/office.actions.ts]: `DID` in the action above (perhaps incorrectly) refers to a lifecycle hook. I'd leave that out here and just go with `ADDED_DANGEROUS_FAULT_COMMENT` (but leaving the page prefix on the value).

File: src/pages/office/office.scss [hunk 2]
```diff
@@ -13,6 +13,13 @@ page-office {
             margin-top: 42px;
           }
         }
+        // ion-textarea {
```
Comment (johnmorleybjss) [inline src/pages/office/office.scss]: To be removed.

File: src/pages/office/office.ts [hunk 3]
```diff
@@ -212,6 +224,12 @@ export class OfficePage extends BasePageComponent {
     ];
   }
 
+  ngAfterViewInit(): void {
+    this.dangerousFaultComment.forEach((comment) => {
+      this.inputChangeSubscriptionDispatchingAction(comment, OfficeViewDidAddDangerousFaultComment);
```
Comment (johnmorleybjss) [inline src/pages/office/office.ts]: You will need to store these subscriptions up somewhere and unsubscribe at some point.

File: src/modules/tests/test-summary/test-summary.selector.ts [hunk 4]
```diff
@@ -18,3 +18,9 @@ export const isDebriefWitnessed = (testSummary: TestSummary): boolean => testSum
 export const isDebriefUnwitnessed = (testSummary: TestSummary): boolean => {
   return (testSummary.debriefWitnessed != null && !testSummary.debriefWitnessed) || false;
 };
+export const debriefWasWitnessed = (testSummary: TestSummary): boolean => {
+  return (testSummary.debriefWitnessed) || false;
+};
+export const debriefWasUnwitnessed = (testSummary: TestSummary): boolean => {
```
Comment (Cavalier07) [inline src/modules/tests/test-summary/test-summary.selector.ts]: This is identical to the isDebriefUnwitnessed function above

Comment (MarkPatt89) [reply src/modules/tests/test-summary/test-summary.selector.ts]: Nice one, Tom. It seems both of these are unused.

**Comment by rossfauldsbjss** [comment]
isDebriefUnwitnessed is used – it’s in the office.ts pagestate.

File: src/modules/tests/test-summary/test-summary.selector.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-summary/test-summary.selector.ts]: It looks like these have come in from different versions.  the correct one is isDebriefUnwitnessed.

Comment (MarkPatt89) [reply src/modules/tests/test-summary/test-summary.selector.ts]: I've removed the offending exports, Ross. Thanks for confirming.

**Pull request merged**